src/modules/kvs/cache.c

```diff
@@ -83,6 +84,25 @@ struct cache_entry *cache_entry_create_json (json_t *o)
     return hp;
 }
 
+struct cache_entry *cache_entry_create_raw (void *data, int len)
+{
+    struct cache_entry *hp;
+
+    if ((data && len <= 0) || (!data && len)) {
```

Comment:
The content cache allows a size = 0 blob to be stored.  Probably we should allow it in the KVS and add test coverage.  Not sure if it worked before, but this would prevent it I guess.

I'll review more closely tomorrow - just noticed that on a quick skim.

Reply:
the intent was if len == 0, data should == NULL.  Would this be ok?

Reply:
Oh duh, sorry, I guess I didn't look closely enough there!

Reply:
I'll open a separate bug on ability to store zero length data under a key as there do seem to be some issues there, unrelated to this PR.